oryx-tui/src/section/inspection.rs

```diff
@@ -657,9 +658,50 @@ impl Inspection {
                 .border_type(BorderType::Thick),
             block,
         );
+
         match app_packet.packet {
-            NetworkPacket::Ip(ip_packet) => ip_packet.render(block, frame),
-            NetworkPacket::Arp(arp_packet) => arp_packet.render(block, frame),
+            NetworkPacket::Ip(ip_packet) => {
+                let (network_packet_block, eth_frame_block) = {
+                    let chunks = Layout::default()
+                        .direction(Direction::Vertical)
+                        .constraints(Constraint::from_percentages([70, 30]))
+                        .flex(ratatui::layout::Flex::SpaceAround)
+                        .margin(2)
+                        .split(block);
+
+                    (chunks[0], chunks[1])
+                };
+
+                // split to apply extra margin to line up with network packet rendering
+                let eth_frame_block = Layout::default()
+                    .direction(Direction::Horizontal)
+                    .constraints(Constraint::from_percentages([95, 5]))
+                    .margin(2)
+                    .split(eth_frame_block)[0];
+
+                EthFrameHeader::from(app_packet.eth_header).render(eth_frame_block, frame);
+                ip_packet.render(network_packet_block, frame)
+            }
+            NetworkPacket::Arp(arp_packet) => {
+                let (arp_block, eth_frame_block) = {
+                    let chunks = Layout::default()
+                        .direction(Direction::Vertical)
+                        .constraints([
+                            Constraint::Fill(1),
+                            Constraint::Percentage(50),
+                            Constraint::Length(6),
+                            Constraint::Fill(1),
+                        ])
+                        .flex(ratatui::layout::Flex::Center)
+                        .margin(2)
+                        .split(block);
+
+                    (chunks[1], chunks[2])
+                };
+
+                arp_packet.render(arp_block, frame);
+                EthFrameHeader::from(app_packet.eth_header).render(eth_frame_block, frame);
+            }
```

Comment:
there is some extra space between the l2 layer and the layers on top that I think this needs some tweaks. I fixed it locally, just let me know if you want me to share them